client/question/services/FeedbackGeneratorService.js

```diff
@@ -402,50 +422,36 @@ tie.factory('FeedbackGeneratorService', [
        */
       getRuntimeErrorFeedback: function(feedbackDetails, rawCodeLineIndexes) {
         var errorInput = feedbackDetails.getErrorInput();
+        var errorLineNumber = feedbackDetails.getErrorLineNumber();
         var errorString = feedbackDetails.getErrorString();
         var language = feedbackDetails.getLanguage();
         var languageUnfamiliarityFeedbackIsNeeded = (
-          feedbackDetails.isLanguageUnfamiliarityFeedbackNeeded());
-
+            feedbackDetails.isLanguageUnfamiliarityFeedbackNeeded());
         var feedback = FeedbackObjectFactory.create(
-          FEEDBACK_CATEGORIES.RUNTIME_ERROR);
-        var fixedErrorString = errorString.replace(
-          new RegExp('line ([0-9]+)$'), function(_, humanReadableLineNumber) {
-            var preprocessedCodeLineIndex = (
-              Number(humanReadableLineNumber) - 1);
-            if (preprocessedCodeLineIndex < 0 ||
-                preprocessedCodeLineIndex >= rawCodeLineIndexes.length) {
-              throw Error(
-                'Line number index out of range: ' + preprocessedCodeLineIndex);
-            }
-            if (rawCodeLineIndexes[preprocessedCodeLineIndex] === null) {
-              console.error(
-                'Runtime error on line ' + preprocessedCodeLineIndex +
-                ' in the preprocessed code');
-              return 'a line in the test code';
-            } else {
-              return 'line ' + (
-                rawCodeLineIndexes[preprocessedCodeLineIndex] + 1);
-            }
+            FEEDBACK_CATEGORIES.RUNTIME_ERROR);
+        if (errorLineNumber !== null) {
+          var preprocessedCodeLineIndex = (Number(errorLineNumber) - 1);
+          if (preprocessedCodeLineIndex < 0 ||
+              preprocessedCodeLineIndex >= rawCodeLineIndexes.length) {
+            throw Error(
+                'Line number index out of range: ' +
+                preprocessedCodeLineIndex);
           }
-        );
-
-        var humanReadableFeedbackString = _getHumanReadableRuntimeFeedback(
-          fixedErrorString, language);
-        if (humanReadableFeedbackString) {
-          feedback.appendTextParagraph(humanReadableFeedbackString);
+        }
+        var friendlyRuntimeFeedbackString = _getFriendlyRuntimeFeedback(
+            errorString, language);
+        if (friendlyRuntimeFeedbackString) {
+          feedback.appendTextParagraph(friendlyRuntimeFeedbackString);
         } else {
           feedback.appendTextParagraph(
-            'Looks like your code had a runtime error when evaluating the ' +
-            'input ' + _jsToHumanReadable(errorInput) + '.');
-          feedback.appendErrorParagraph(fixedErrorString);
+              'Looks like your code had a runtime error when evaluating the ' +
```

Comment:
Here and elsewhere -- I understand that indentation by 4 is idiomatic in some styles, but most of the codebase goes with 2-space indentation for everything. I think we should either fix indentation in the whole codebase or keep the existing convention throughout, rather than end up with a half-and-half.